### PR TITLE
Fix links to configuration page

### PR DESCRIPTION
**Pull request by fjvela**
**What this PR does / why we need it**:
Fix links to configuration page:
- kafka
- Cloudflare
- Kubernetes Discovery
- Windows Event Log
- Heroku Drain
- HTTP client options

### CODE REVIEW

File: docs/sources/clients/promtail/scraping.md
Comment (DylanGuedes) [inline docs/sources/clients/promtail/scraping.md:376]: We should avoid using hardcoded links.  Don't `[configuration](./configuration/#kafka)` or just `[configuration](#kafka)` work here?

File: docs/sources/clients/promtail/scraping.md
Comment (DylanGuedes) [inline docs/sources/clients/promtail/scraping.md:425]: ditto

**Review by DylanGuedes — CHANGES_REQUESTED** [review]
Hey thanks for your review! I think we should instead look for an alternative that doesn't rely on the hardcoded links, otherwise users will be redirected to the latest version even when using an older one.

**Review by osg-grafana — DISMISSED** [review]
Needs guidance from @jdbaldry who is handling doc links across all Grafana Labs repos

File: docs/sources/clients/promtail/scraping.md
Comment (jdbaldry) [reply docs/sources/clients/promtail/scraping.md:376]: You should be able to see docs if you browse to localhost:3002/docs/loki/latest (https://github.com/grafana/loki/blob/0da555385ab87ba7a535050979950ecffb0be792/docs/Makefile#L9)

File: docs/sources/clients/promtail/scraping.md
Comment (jdbaldry) [reply docs/sources/clients/promtail/scraping.md:425]: Yeah using Hugo relrefs (https://gohugo.io/functions/relref/) would be the best solution. The links will be checked when you run `make docs`

Comment (jdbaldry) [reply docs/sources/clients/promtail/scraping.md:425]: Additionally if we add https://github.com/grafana/loki/pull/6569 to CI, you can get anchor checking too

File: docs/sources/clients/promtail/scraping.md
Comment (fjvela) [reply docs/sources/clients/promtail/scraping.md:376]: Thanks!

File: docs/sources/clients/promtail/scraping.md
Comment (fjvela) [reply docs/sources/clients/promtail/scraping.md]: Thanks.

**Comment by fjvela** [comment]
@jdbaldry @osg-grafana @DylanGuedes  can you take a look?

File: docs/sources/clients/promtail/scraping.md
Comment (jdbaldry) [inline docs/sources/clients/promtail/scraping.md:542]: nit: Not related to your change but I'm seeing that the anchor here doesn't exist in the linked file:
```console
$ docker run -it -v "$(pwd)/docs/sources:/docs/sources" grafana/doc-validator:latest /docs/sources 2>&1 | grep -A1 docs/sources/clients/promtail/scraping.md
ERROR: /docs/sources/clients/promtail/scraping.md:542:28 The anchor "client_config" does not exist in the file "/docs/sources/clients/promtail/configuration.md".
Available anchors are: "configuring-promtail, printing-promtail-config-at-runtime, configuration-file-reference, use-environment-variables-in-the-configuration, generic-placeholders, supported-contents-and-default-values-of-configyaml, server, clients, positions, scrape_configs, pipeline_stages, docker, cri, regex, json, template, match, timestamp, output, labels, metrics, counter, gauge, histogram, tenant, replace, journal, syslog, available-labels, loki_push_api, windows_events, gcp-log, available-labels, kafka, gelf, cloudflare, heroku_drain, available-labels, relabel_configs, static_configs, file_sd_config, kubernetes_sd_config, node, service, pod, endpoints, ingress, consul_sd_config, consulagent_sd_config, docker_sd_config, limits_config, target_config, options_config, tracing_config, example-docker-config, example-static-config, example-static-config-without-targets, example-journal-config, example-syslog-config, example-push-config".
```

**Comment by jdbaldry** [comment]
Apologies for the delay in reviewing this. I really appreciate the fixes to the links because now we can have link checking in our local development.

I've added one nit and will let @DylanGuedes and @osg-grafana re-review so they can remove their request for changes if they are happy and we can get this merged.

One note that we should just double check. Given the age of this branch, it might make sense to sync with `main`, either by rebase or merge, so that we can ensure nothing has moved around under us.

**Review by DylanGuedes — APPROVED** [review]
ack, I'm approving to remove my "request changes" as you addressed them (thank you). However, I didn't test the new links so would be cool if someone else could check them before the merge.

File: docs/sources/clients/promtail/scraping.md
Comment (fjvela) [reply docs/sources/clients/promtail/scraping.md:542]: I just checked it and It doesn't exits but I don't see any other anchor related to link it and fix it.

Comment (jdbaldry) [reply docs/sources/clients/promtail/scraping.md:542]: I just took a look through the target page. I think `clients` is the heading we want as it describes an array of `<client_config>`

> # Describes how Promtail connects to multiple instances
> # of Grafana Loki, sending logs to each.
> # WARNING: If one of the remote Loki servers fails to respond or responds
> # with any error which is retryable, this will impact sending logs to any
> # other configured remote Loki servers.  Sending is done on a single thread!
> # It is generally recommended to run multiple Promtail clients in parallel
> # if you want to send to multiple remote Loki instances.
> clients:
>   - [<client_config>]

**Comment by jdbaldry** [comment]
@osg-grafana would you mind re-reviewing this for your requested changes?

**Comment by jdbaldry** [comment]
I've dismissed @osg-grafana's request for changes as I believe they have been addressed and they are currently OOO until the new year.

@grafana/loki-squad is anyone able to merge this?

The CI failure appears to be unrelated.

**Review by MichelHollands — APPROVED** [review]
LGTM

**Comment by fjvela** [comment]
Crazy... I don't know why but some of them doesn't work... (facepalm)

**Pull request merged**
